Feat: testing class added

oh0227 opened the pull request:
게임 스크린에서 기본적으로 일어나는 상황들에 대한 테스트 코드 작성

oh0227 committed:
feat: JUnit api added

JUnit api를 gradle build에 추가함.

oh0227 committed:
feat: bullet testing class added

Bullet Class를 테스트하는 BulletTest class를 생성

oh0227 committed:
Remove: bullet testing class removed

Bullet Class를 테스트하는 BulletTest class를 제거(의미없는 테스트)

oh0227 committed:
Feat: GameScreen testing class added

GameScreen의 initialize와 update 메소드를 테스트함으로써 게임구동에 지장이 없음을 확인한다.

oh0227 committed:
Feat: lombok api added

테스트 코드에 필요한 변수만 @Getter를 통해 가져옴

oh0227 committed:
Feat: collisions testing code added

객체끼리 닿았을 때의 상황이 잘 처리되는지 확인하는 테스트 코드 추가

oh0227 committed:
Fix: update testing code

update 메소드 테스트가 무한 루프를 돌던 상황을 일정 시간 이상 지나면 테스트가 실패되도록 변경,
물체가 총을 맞아서 죽는 경우 SpriteType이 제대로 변하는지 학인하는 기능 추가,
manageCollision 테스트를 통과하지 못하면 아예 update 테스트를 못하도록 설정

ag-0224 reviewed (approved):
확인했습니다. 고생하셨습니다.

jeonjw95 commented on src/engine/DrawManager.java:
```diff
@@ -149,7 +149,7 @@ public void initDrawing(final Screen screen) {
 				BufferedImage.TYPE_INT_RGB);
 
-		graphics = frame.getGraphics();
+		if(frame != null) graphics = frame.getGraphics();
```
해당 라인이 변경된 이유가 궁금합니다

oznwoo replied:
게임스크린을 테스트할 때는 화면에 창을 띄울 필요 없이 테스트만 진행하기 위해서 추가한 코드입니다.

jeonjw95 replied:
실제 게임에는 영향이 없는건가요?

oznwoo replied:
네 실제 게임에는 영향 없습니다!

jeonjw95 commented on src/test/java/screen/GameScreenTest.java:
```diff
@@ -0,0 +1,215 @@
+      gameScreen.update();
+      // 10초 이내에 루프를 빠져나오도록 확인
+      if (System.currentTimeMillis() - loopStartTime > loopDuration) {
+        fail("루프가 10초 이상 실행됨"); // 테스트 실패로 처리하거나 다른 적절한 조치를 취할 수 있습니다.
```
10초 이상 실행됐다는 것이 이 테스트의 실패와 연관이 없는 것 같습니다. 무한루프를 방지하고 테스트 시간을 줄일 수 있는 방법을 모색하면 좋을 것 같습니다.

oznwoo replied:
새로 commit한 내용에 수정했습니다!

jeonjw95 commented on src/test/java/screen/GameScreenTest.java:
```diff
@@ -0,0 +1,215 @@
+      gameScreen.update();
+      // 10초 이내에 루프를 빠져나오도록 확인
+      if (System.currentTimeMillis() - loopStartTime > loopDuration) {
+        fail("루프가 10초 이상 실행됨"); // 테스트 실패로 처리하거나 다른 적절한 조치를 취할 수 있습니다.
```
위 리뷰와 동일합니다

msk226 commented on src/test/java/screen/GameScreenTest.java:
```diff
@@ -0,0 +1,215 @@
+      gameScreen.update();
+      // 10초 이내에 루프를 빠져나오도록 확인
+      if (System.currentTimeMillis() - loopStartTime > loopDuration) {
+        fail("루프가 10초 이상 실행됨"); // 테스트 실패로 처리하거나 다른 적절한 조치를 취할 수 있습니다.
```
159번 라인에 있는 설명에는 15초 이상 실행하면 강제 종료한다고 되어있는데, 지금 코드 상에는 "10초 이상 실행 됨" 이라고 나와있어서 모호한 것 같습니다.

oznwoo replied:
새로 commit한 내용에 수정했습니다!

ssxrxbx reviewed (approved):
팀원 분들의 리뷰 말고는 이의 없습니다. 고생하셨습니다!

oh0227 committed:
Fix: apply code review

1. 코드 리뷰에서 언급된 무한루프 방지를 위한 조건 및 테스트 추가
2. 기존에 무한루프 방지를 위해 설정한 시간 제한 재거
3. EnemyShipFormation, GameScreen 클래스에 테스팅 할때만 활성화되는 변수 추가

oh0227 committed:
Fix: testing condition added

이전 테스트를 성공해야 다음 테스트를 수행할 수 있도록 설정하여 무한루프에 빠지는 일이 거의 없도록 설정하였다.

oh0227 committed:
Merge branch 'develop' into feature/SmokeTest

oh0227 committed:
Fix: fix comments typo

주석 설명 수정

oh0227 committed:
Merge remote-tracking branch 'origin/feature/SmokeTest' into feature/SmokeTest

oh0227 committed:
Fix: clarify testing methods

테스트 메소드들의 이름과 분류 형태를 변경하여 더 직관적으로 테스트 결과를 확인할 수 있도록 함.

jeonjw95 commented on src/test/java/screen/GameScreenTest.java:
```diff
@@ -0,0 +1,288 @@
+
+      }
+    }
+    private static boolean test0() {return test0;}
```
static이 필요한 코드인가요? 필요하다면 public으로 변경하고 필요없다면 static을 제거 바랍니다

oznwoo replied:
없어도 큰 지장이 없는 코드라 새 commit에서 삭제했습니다!

jeonjw95 reviewed (changes requested):
gradle build 문제가 발생합니다. 해결바랍니다!

error: package org.junit.jupiter.api does not exist import static org.junit.jupiter.api.Assertions.*;

oh0227 committed:
Fix: apply code review

코드 리뷰에서 언급된 메소드 제거

oh0227 committed:
Fix: build.gradle fixed

gradle 환경에서 테스트 코드가 실행되지 않던 문제 해결을 위해 build.gradle 파일 수정

oh0227 committed:
Fix: build.gradle fixed

gradle 환경에서 테스트 코드가 빌드되지 않던 문제 해결을 위해 build.gradle 파일 수정

oh0227 commented:
> gradle build 문제가 발생합니다. 해결바랍니다!
> 
> error: package org.junit.jupiter.api does not exist import static org.junit.jupiter.api.Assertions.*;

해결했습니다. 잘 되는지 확인 부탁드립니다!

ag-0224 reviewed (approved):
gradle 수정 사항 확인했습니다. 고생많으셨습니다.
